### PR TITLE
Fixed intersectMany function

### PR DESCRIPTION
**Pull request by matteoscaramuzza**
This change fixes issue #159.
The previous intersectMany method was not finding the correct occurrences.

### CODE REVIEW

**Comment by matteoscaramuzza** [comment]
Done 👍🏻

Let me know if you need anything else.

Grazie!

**Review by micheleriva — CHANGES_REQUESTED** [review]
Left a couple of comments, could you please add tests on array intersection specifically?
Something like:

```js
const arr1 = [1,2,3]
const arr2 = [2,3,4]

const intersection = intersectMany(arr1, arr2) // 2,3
```

thanks!

**Comment by matteoscaramuzza** [comment]
I've just added the requested test.
I've also create the type alias and put it inside the datasets folder.

thanks!

File: tests/datasets/event-json.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+export type EventJson = {
```
Comment (micheleriva) [inline tests/datasets/event-json.ts:1]: A couple of things:

1) Type definitions should be inside *.d.ts files, as a general convention
2) This type alias is not used in other script files so I'd just put this at the top of the only test file where it's used, if really needed

File: tests/utils.test.ts [hunk 2]
```diff
@@ -1,8 +1,21 @@
...
+  t.plan(3);
+
+  t.test("should correctly intersect 2 or more arrays", t => {
+    t.plan(4);
+
+    const arr1 = [1, 2, 3, 4, 5, 8];
+    const arr2 = [2, 3, 8];
+    const arr3 = [4, 6, 5, 8];
+
+    t.equal(intersectMany([arr1, arr2]).length, 3);
```
Comment (micheleriva) [inline tests/utils.test.ts:14]: I wouldn't test length but the actual resulting arrays instead

**Review by micheleriva — CHANGES_REQUESTED** [review]
I left a couple of comments, remember to run the linter before committing 🙂

Thanks again!

File: tests/datasets/event-json.ts (hunk 1 above)
Comment (matteoscaramuzza) [reply tests/datasets/event-json.ts]: Done 👍🏻

**Review by micheleriva — APPROVED** [review]
LGTM thanks!

**Pull request merged**